packages/block-editor/src/components/grid-visualizer/grid-item-resizer.js

```diff
@@ -12,14 +13,89 @@ import { getComputedCSS } from './utils';
 
 export function GridItemResizer( { clientId, onChange } ) {
 	const blockElement = useBlockElement( clientId );
+	const rootBlockElement = blockElement?.parentElement;
+	const [ resizeDirection, setResizeDirection ] = useState( null );
+
+	/*
+	 * Resizer dummy is an empty div that exists only so we can
+	 * get the bounding client rect of the resizer element. This is
+	 * necessary because the resizer exists outside of the iframe, so
+	 * its bounding client rect isn't the same as the block element's.
+	 */
```

Comment:
Could we use `ref` on the `ResizableBox`? Or `__unstableContentRef` on the `BlockPopover`? Would love to avoid more DOM nodes if possible.

Reply:
I don't think we can use `ref` on resizable box because we can't forward it to the markup in the third party component. Let me try with BlockPopover though.

Reply:
We should be able to make it work with `ResizableBox`. The problem is just that right now `ResizableBox` says that it takes a `ref: ForwardedRef< Resizable >` but actually it only accepts `ref: ( c: Resizable ) => void`.

If we update the component so that it works how its type says it works:

```diff
diff --git a/packages/components/src/resizable-box/index.tsx b/packages/components/src/resizable-box/index.tsx
index 0955c29c53..b5505d212b 100644
--- a/packages/components/src/resizable-box/index.tsx
+++ b/packages/components/src/resizable-box/index.tsx
@@ -114,7 +114,15 @@ function UnforwardedResizableBox(
                        ) }
                        handleClasses={ HANDLE_CLASSES }
                        handleStyles={ HANDLE_STYLES }
-                       ref={ ref }
+                       ref={ ( component ) => {
+                               if ( ref ) {
+                                       if ( typeof ref === 'function' ) {
+                                               ref( component );
+                                       } else {
+                                               ref.current = component;
+                                       }
+                               }
+                       } }
                        { ...props }
                >
                        { children }
```

Then we can access the `resizable` attribute of `Resizable` to get at the element:

```diff
diff --git a/packages/block-editor/src/components/grid-visualizer/grid-item-resizer.js b/packages/block-editor/src/components/grid-visualizer/grid-item-resizer.js
index c18cd590d3..58035105b5 100644
--- a/packages/block-editor/src/components/grid-visualizer/grid-item-resizer.js
+++ b/packages/block-editor/src/components/grid-visualizer/grid-item-resizer.js
@@ -15,14 +15,7 @@ export function GridItemResizer( { clientId, onChange } ) {
        const blockElement = useBlockElement( clientId );
        const rootBlockElement = blockElement?.parentElement;
        const [ resizeDirection, setResizeDirection ] = useState( null );
-
-       /*
-        * Resizer dummy is an empty div that exists only so we can
-        * get the bounding client rect of the resizer element. This is
-        * necessary because the resizer exists outside of the iframe, so
-        * its bounding client rect isn't the same as the block element's.
-        */
-       const resizerDummyRef = useRef( null );
+       const resizerRef = useRef();

        if ( ! blockElement ) {
                return null;
@@ -63,7 +56,7 @@ export function GridItemResizer( { clientId, onChange } ) {
                offsetHeight: rootBlockElement.offsetHeight,
                getBoundingClientRect: () => {
                        const resizerTop =
-                               resizerDummyRef.current?.getBoundingClientRect()?.top;
+                               resizerRef.current?.resizable.getBoundingClientRect()?.top;
                        // Fallback value of 60 to account for editor top bar height.
                        const heightDifference = resizerTop
                                ? resizerTop - blockClientRect.top
@@ -98,6 +91,7 @@ export function GridItemResizer( { clientId, onChange } ) {
                        additionalStyles={ styles }
                >
                        <ResizableBox
+                               ref={ resizerRef }
                                className="block-editor-grid-item-resizer__box"
                                size={ {
                                        width: '100%',
@@ -179,10 +173,6 @@ export function GridItemResizer( { clientId, onChange } ) {
                                        } );
                                } }
                        />
-                       <div
-                               className="block-editor-grid-item-resizer__dummy"
-                               ref={ resizerDummyRef }
-                       ></div>
                </BlockPopoverCover>
        );
 }
```

Reply:
Oooh I see. `__unstableContentRef` seems to work pretty well though, and it's a tiny change so I think I'll just go with that for now. Would be good to also fix the ref in `ResizableBox` but maybe in its own PR?